src/renderer/components/+extensions/attempt-install/attempt-install.tsx

```diff
@@ -130,5 +101,9 @@ export const attemptInstall =
             onClose: dispose,
           },
         );
+      } else {
+        // Remove the old dir because it isn't a valid extension anyway
+        await removeDir(extensionFolder);
+        await unpackExtension(validatedRequest, dispose);
```

Comment:
This is the main substance of the bug fix. Namely, instead of assuming that the folder existing is the same as the extension being installed, we check if there is an installed extension and if not we delete all the old data just in case.

Reply:
Can this be 1 PR that fixes the bug for 5.4 and the rest be another PR for 5.5?

Reply:
Sure can do, will do that tomorrow